notify.js

```diff
@@ -139,16 +153,30 @@
             'requireInteraction': this.options.requireInteraction,
             'silent': this.options.silent,
             'closeOnClick': this.options.closeOnClick
-        });
+        };
 
-        if (!this.options.requireInteraction && this.options.timeout && !isNaN(this.options.timeout)) {
-            setTimeout(this.close.bind(this), this.options.timeout * 1000);
-        }
+        if (this.options.serviceWorker) {
+            var self = this;
+
+            this.options.serviceWorker.ready.then(function(registration) {
+                registration.showNotification(title, options)
+                .then(function() {
+                    return registration.getNotifications();
+                })
+                .then(function(notifications) {
+                    self.myNotify = notifications[notifications.length - 1]; // TODO: Naive implementation, improve
+                    bindNotificationEvents(self);
```

Comment:
The thing with MDN is that there are some misleading things there. For example:

https://developer.mozilla.org/en-US/docs/Web/API/NotificationEvent/notification

Says that `NotificationEvent.notification` returns a `Notification` object:

https://developer.mozilla.org/en-US/docs/Web/API/Notification

However, the events supported in the `Notification` event do not seem to fire.
Another misleading thing was this:
https://developer.mozilla.org/en-US/docs/Web/API/ServiceWorkerRegistration/showNotification

MDN says that the method returns a `Promise` that resolves to a `NotificationEvent`, however, that is not true, it resolves to `undefined`, and if you look at the actual specification here:

https://notifications.spec.whatwg.org/#service-worker-api

You see that the `Promise` resolves to `void`
You basically need to test these things one by one, can't rely on the documentation